### PR TITLE
Adds basic tools and structures

### LINKED ISSUE #8: Add code needed for basic test distributions in a differentiable way (e.g two moons, roll)
This is issue is to discuss which distributions we want to implement and to setup a basic package. I'll be working on it on this branch `basic_infra`.

### PR DESCRIPTION
**Pull request by EiffL**
This PR adds the basic structure of python package and adds two toy distributions for us to play with. 

I have also prepared a demo notebook, [here](https://colab.research.google.com/drive/1rxzrg5hek6TfHaV-z-VmNvGaQdwnaDx7?usp=sharing) that I wanted to add to the PR but can't commit directly from colab yet. @aboucaud if you are the owner can you grant my colab access request to the orga? 

This PR aims to close #8

### CODE REVIEW

**Comment by aboucaud** [comment]
Access granted !

**Comment by aboucaud** [comment]
Waiting for the notebook to review

**Comment by EiffL** [comment]
Ready for review!

File: sbiexpt/distributions.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  ----------
+  sigma: float
+  Spread of the 2 moons distribution.
+  resolution: int
+  Number of components in the gaussian mixture approximation of the
+  distribution (default: 1024)
+  Returns
+  -------
+  distribution: TFP distribution
+  Two moon distribution
```
Comment (aboucaud) [inline sbiexpt/distributions.py:34]:
```suggestion
  Returns two moons distribution as a TFP distribution
  
  Parameters
  ----------
  sigma: float
    Spread of the 2 moons distribution.
  resolution: int
    Number of components in the gaussian mixture approximation of the
    distribution (default: 1024)
  
  Returns
  -------
  distribution: TFP distribution
    Two moons distribution
```

Comment (aboucaud) [reply sbiexpt/distributions.py:34]: Just being picky :)

**Review by aboucaud — COMMENTED** [review]
Looking good

File: sbiexpt/distributions.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import jax
+import jax.numpy as np
+from sklearn.datasets import make_swiss_roll
+import tensorflow_probability as tfp; tfp = tfp.experimental.substrates.jax
```
[the remark this answers is not in the document]
Comment (EiffL) [reply sbiexpt/distributions.py:4]: Indeed!

File: sbiexpt/distributions.py (hunk 1 above)
Comment (EiffL) [reply sbiexpt/distributions.py:34]: ^^' no problem with that, especially since you provided the fix

**Pull request merged**